src/docker/custom.rs

```diff
@@ -278,7 +278,11 @@ fn docker_tag_name(file_name: &str) -> String {
         }
     }
 
-    // in case all characters were invalid, use a non-empty filename
+    // in case all characters were invalid or we had all non-ASCII
+    // characters followed by a `-`, we use a non-empty filename
+    if result.ends_with('-') {
```

Comment:
Actually this can now be empty, if we have something like `한글-`, which is available on nightly.

Reply:
maybe converting to ascii via `let file_name = file_name.as_bytes().escape_ascii()` would work. Making it into a punycode-ish non-unique string

Reply:
I think it's fine already since the custom image name depends on the target and the path hash. The name is just to make it user-friendly: if we've got Unicode characters, punycode-ish isn't very [friendly](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=11679b7219273da5d8000a9eb99427f5).

```rust
pub fn main() {    
    let x = "cargo한글-";
    println!("escape_ascii={:?}", x.as_bytes().escape_ascii().to_string());
}
```

```bash
$ rustc main.rs
$ ./main
escape_ascii="cargo\\xed\\x95\\x9c\\xea\\xb8\\x80-"
```

After creating the tag name, we'd get something like `cargoxedx95x9cxeaxb8x80`, which I don't think is very helpful. The Docker image would already be named `localhost/cross-rs/cross-custom-empty:armv7-unknown-linux-gnueabihf-ec99f-pre-build`, where `ec99f` is the path hash. If we were worried about collisions (~1/1000)`, extending the path hash to 10 characters might be better.